orbit/pkg/update/hash.go

```diff
@@ -12,29 +12,34 @@ import (
 	"github.com/theupdateframework/go-tuf/data"
 )
 
-// CheckFileHash checks the file at the local path against the provided hash
-// functions.
-func CheckFileHash(meta *data.TargetFileMeta, localPath string) error {
-	hashFunc, hashVal, err := selectHashFunction(meta)
+// checkFileHash checks the file at the local path against the provided hash functions.
+func checkFileHash(meta *data.TargetFileMeta, localPath string) error {
+	localHash, metaHash, err := fileHashes(meta, localPath)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to calculate local file hash: %s", err)
 	}
+	if !bytes.Equal(localHash, metaHash) {
+		return fmt.Errorf("hash %s does not match expected: %s", data.HexBytes(localHash), data.HexBytes(metaHash))
```

Comment:
Minor, but you can use the `%x` (or `%X` for uppercase) formatting verbs to print the hex bytes of a byte slice, without manipulating the data (https://pkg.go.dev/fmt#hdr-Printing).